core/iwasm/aot/aot_runtime.c

```diff
@@ -2431,6 +2431,55 @@ aot_call_indirect(WASMExecEnv *exec_env, uint32 tbl_idx, uint32 table_elem_idx,
     }
 }
 
+/**
+ * Check whether the app address and the buf is inside the linear memory,
+ * and convert the app address into native address
+ */
+bool
+aot_check_app_addr_and_convert(AOTModuleInstance *module_inst, bool is_str_arg,
+                               uint32 app_offset, uint32 buf_size,
+                               void **p_native_addr)
+{
```

Comment:
how about renaming parameters?
- `is_str_ag` -> `is_str_buf`
- `app_offset` -> `buf_offset` or `buf_mem_offset`
- `p_native_addr` -> `buf_native_addr`

Reply:
The function is somewhat like other APIs:
```
bool
wasm_runtime_validate_app_addr(WASMModuleInstanceCommon *module_inst,
                               uint32 app_offset, uint32 size);
bool
wasm_runtime_validate_native_addr(WASMModuleInstanceCommon *module_inst,
                                  void *native_ptr, uint32 size);
uint32
wasm_runtime_addr_native_to_app(WASMModuleInstanceCommon *module_inst,
                                void *native_ptr);
```
app_xxx means relative address of wasm app, native_xxx mean absolute address of native,
how about changing to:
``` 
bool
aot_check_app_addr_and_convert(AOTModuleInstance *module_inst, bool is_str_buf,
                               uint32 app_buf_addr, uint32 app_buf_size,
                               void **p_native_buf_addr);
```

Reply:
got the point now. 

```
bool aot_check_app_addr_and_convert(AOTModuleInstance *module_inst, 
     bool is_str, uint32 app_addr, uint32 size_or_len, void **p_native_addr);
```